syntax="proto3"; // Kullandığımız ProtoBuf versiyonu

package company;

// görev durumunu tutacağımız enum sabiti
enum State{
    PLANNED=0;
    IN_PROGRESS=1;
    DONE=2;
}

// ana tipimiz Job
// içinde n sayıda Task içerebilir ki o da bir sözleşme tipidir
// niteliklere atanan sayılar binary encode edilen çıktıdaki alanları işaret eden benzersiz sayılardır. 
// Bir nevi eşleştirme veya yer bulma tekniği.
message Job{    
    int32 job_id=1;
    string title=2;

    // İşteki görevleri tanımlayan tipimiz
    message Task{
        int32 task_id=1;
        string title=2;
        string description=3;
        State state=4; //enum sabiti tipinden
    }

    repeated Task tasks=3; //Task tipinin tekrarlanabileceğini belirtiyor
}